packages/block-library/src/container/index.js

```diff
@@ -0,0 +1,159 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { withFallbackStyles } from '@wordpress/components';
+import { compose } from '@wordpress/compose';
+import { __, sprintf } from '@wordpress/i18n';
+import {
+	getColorClassName,
+	withColors,
+	ContrastChecker,
+	InnerBlocks,
+	InspectorControls,
+	PanelColorSettings,
+} from '@wordpress/editor';
+import { Fragment } from '@wordpress/element';
+
+const applyFallbackStyles = withFallbackStyles( ( node, ownProps ) => {
+	const { textColor, backgroundColor } = ownProps.attributes;
+
+	return {
+		fallbackBackgroundColor: backgroundColor,
+		fallbackTextColor: textColor,
+	};
+} );
+
+export const name = 'core/container';
+
+export const settings = {
+	title: sprintf(
+		/* translators: Block title modifier */
+		__( '%1$s (%2$s)' ),
+		__( 'Container' ),
+		__( 'beta' )
+	),
+
+	icon: <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path d="M19 12h-2v3h-3v2h5v-5zM7 9h3V7H5v5h2V9zm14-6H3c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h18c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16.01H3V4.99h18v14.02z" /><path d="M0 0h24v24H0z" fill="none" /></svg>,
+
+	category: 'layout',
+
+	attributes: {
+		align: {
+			type: 'string',
+		},
+		backgroundColor: {
+			type: 'string',
+		},
+		customBackgroundColor: {
+			type: 'string',
+		},
+		textColor: {
+			type: 'string',
+		},
+		customTextColor: {
+			type: 'string',
+		},
+	},
+
+	description: __( 'Group blocks into a container.' ),
+
+	supports: {
+		align: [ 'wide', 'full' ],
+		anchor: true,
+	},
+
+	edit: compose( [
+		withColors( 'backgroundColor', { textColor: 'color' } ),
+		applyFallbackStyles,
+	] )( ( props ) => {
+		const {
+			backgroundColor,
+			className,
+			fallbackBackgroundColor,
+			fallbackTextColor,
+			setBackgroundColor,
+			setTextColor,
+			textColor,
+		} = props;
+
+		return (
+			<Fragment>
+				<InspectorControls>
+					<PanelColorSettings
+						title={ __( 'Color Settings' ) }
+						initialOpen={ false }
+						colorSettings={ [
+							{
+								value: backgroundColor.color,
+								onChange: setBackgroundColor,
+								label: __( 'Background Color' ),
+							},
+							{
+								value: textColor.color,
+								onChange: setTextColor,
+								label: __( 'Text Color' ),
+							},
+						] }
+					>
+						<ContrastChecker
+							{ ...{
+								textColor: textColor.color,
+								backgroundColor: backgroundColor.color,
+								fallbackTextColor,
+								fallbackBackgroundColor,
+							} }
+						/>
+					</PanelColorSettings>
+				</InspectorControls>
+				<div
+					className={ classnames( className, {
+						'has-background': backgroundColor.color,
+						'has-text-color': textColor.color,
+						[ backgroundColor.class ]: backgroundColor.class,
+						[ textColor.class ]: textColor.class,
+					} ) }
+					style={ {
+						backgroundColor: backgroundColor.color,
```

Comment:
Should the inline styles be necessary? Why isn't it enough to assign the color classes?

(Maybe this is a known limitation, but at a high-level it feels wrong)

Reply:
@aduth if someone picks a custom color, it has to be applied via `style`. I just copied this over from the paragraph block.

Reply:
So then what's the purpose of the class name being assigned?

Again, not particularly directed at you or this block, maybe more a general issue we have with colors assignment.

Reply:
@aduth If you pick one of the colors in the palette, it assigns a class. If you pick a color outside the palette, it applies a style.

This is the rendered HTML for a block using the default red/light gray colors:

```html
<div class="wp-block-container has-background has-text-color has-vivid-red-background-color has-very-light-gray-color">
<p>Container, Paragraph One</p>
<p>Container, Paragraph Two</p>
</div>
```

Vs this rendered HTML from the same block using two custom colors selected via the color picker popover:

```
<div class="wp-block-container has-background has-text-color" style="background-color:#2eaacf;color:#ffffff">
<p>Container, Paragraph One</p>
<p>Container, Paragraph Two</p>
</div>
```

I'm not quite sure how this magic works, but it does 😅

Reply:
@aduth Oh I see you're referring to the edit representation of the block, not the front-end version, and that's a great question. I'll take a closer look.

Reply:
I'm pushing up a commit to fix this. I can open a separate PR to do this for the paragraph block as well.